monitor/src/main/java/edu/asu/diging/monitor/web/admin/AddRecipientController.java

```diff
@@ -21,12 +25,22 @@
 
 	@Autowired
 	private INotificationManager manager;
+	
+	@Autowired
+	private IAppManager appManager;
 
-	@RequestMapping(value = "/admin/recipients/add", method = RequestMethod.GET)
-	public String show(Model model) {
-		model.addAttribute("recipientForm", new RecipientForm());
-		return "admin/recipients/add";
-	}
+    @RequestMapping(value = "/admin/recipients/add", method = RequestMethod.GET)
+    public String show(Model model) {
+        RecipientForm recipientForm = new RecipientForm();
+        recipientForm.setApps(appManager.getApps().stream().map(x -> {
```

Comment:
`x` is a pretty bad variable name. even in a lambda function, it should be somewhat related to the object e.g. `a` for app